hardhat.config.ts

```diff
@@ -50,396 +46,383 @@ const config: HardhatUserConfig = {
       url: "http://127.0.0.1:8545",
     },
     tenderly: {
-      url: `https://rpc.tenderly.co/fork/${process.env.TENDERLY_FORK_ID}`,
+      url: `https://rpc.tenderly.co/fork/${
+        vars.has("TENDERLY_FORK_ID") ? vars.get("TENDERLY_FORK_ID") : ""
+      }`,
     },
     devnet: {
-      url: `https://rpc.vnet.tenderly.co/devnet/${process.env.TENDERLY_DEVNET_ID}`,
-      accounts:
-        process.env.PRIVATE_KEY !== undefined ? [process.env.PRIVATE_KEY] : [],
+      url: `https://rpc.vnet.tenderly.co/devnet/${
+        vars.has("TENDERLY_DEVNET_ID") ? vars.get("TENDERLY_FORK_ID") : ""
+      } `,
+      accounts: vars.has("PRIVATE_KEY") ? [vars.get("PRIVATE_KEY")] : [],
     },
     goerli: {
       chainId: 5,
-      url: process.env.ETH_GOERLI_TESTNET_URL || "",
-      accounts:
-        process.env.PRIVATE_KEY !== undefined ? [process.env.PRIVATE_KEY] : [],
+      url: vars.has("ETH_GOERLI_TESTNET_URL")
+        ? vars.get("ETH_GOERLI_TESTNET_URL")
+        : "",
+      accounts: vars.has("PRIVATE_KEY") ? [vars.get("PRIVATE_KEY")] : [],
     },
     sepolia: {
       chainId: 11155111,
-      url: process.env.ETH_SEPOLIA_TESTNET_URL || "",
-      accounts:
-        process.env.PRIVATE_KEY !== undefined ? [process.env.PRIVATE_KEY] : [],
+      url: vars.has("ETH_SEPOLIA_TESTNET_URL")
+        ? vars.get("ETH_SEPOLIA_TESTNET_URL")
+        : "",
+      accounts: vars.has("PRIVATE_KEY") ? [vars.get("PRIVATE_KEY")] : [],
     },
     holesky: {
       chainId: 17000,
-      url: process.env.ETH_HOLESKY_TESTNET_URL || "",
-      accounts:
-        process.env.PRIVATE_KEY !== undefined ? [process.env.PRIVATE_KEY] : [],
+      url: vars.has("ETH_HOLESKY_TESTNET_URL")
+        ? vars.get("ETH_HOLESKY_TESTNET_URL")
+        : "",
+      accounts: vars.has("PRIVATE_KEY") ? [vars.get("PRIVATE_KEY")] : [],
     },
     ethMain: {
       chainId: 1,
-      url: process.env.ETH_MAINNET_URL || "",
-      accounts:
-        process.env.PRIVATE_KEY !== undefined ? [process.env.PRIVATE_KEY] : [],
+      url: vars.has("ETH_MAINNET_URL") ? vars.get("ETH_MAINNET_URL") : "",
+      accounts: vars.has("PRIVATE_KEY") ? [vars.get("PRIVATE_KEY")] : [],
```

Comment:
Should we extract this into a constant to dedupe it? Also wonder if we should have a helper function to avoid copy/paste errors like the tenderly one
```typescript
const accounts = vars.has("PRIVATE_KEY") ? [vars.get("PRIVATE_KEY")] : [];
const rpcUrl = (envVarName: string) => vars.has(envVarName) ? vars.get() : "";

// --- snip ---

    sepolia: {
      chainId: 11155111,
      url: rpcUrl("ETH_SEPOLIA_TESTNET_URL"),
      accounts,
    },
    holesky: {
      chainId: 17000,
      url: rpcUrl("ETH_HOLESKY_TESTNET_URL"),
      accounts,
    },
```

Reply:
I adjusted it in a way that all URLs now have a backup RPC URL & I checked each of them for validity. Also, I simplified the `accounts` value as suggested. For all this I used the default value param of `vars.get` which simplifies this a lot.